ts/main.ts

```diff
@@ -256,3 +266,44 @@ $confirmDeleteBtn.addEventListener('click', () => {
   viewSwap('entries');
   resetEntryForm();
 });
+
+$searchForm.addEventListener('submit', (event: Event) => {
+  event.preventDefault();
+
+  const searchTerm = $searchInput.value.toLowerCase();
+
+  if (!searchTerm) return;
+
+  $searchInput.className = 'with-clear';
+  $clearSearchBtn.className = '';
+
+  for (let i = 0; i < data.entries.length; i++) {
+    const $currentLi = document.querySelector(
+      `li[data-entry-id="${data.entries[i].entryId}"]`,
+    );
+    if (!$currentLi) throw new Error('$hideLi query failed');
+    if (
+      !data.entries[i].title.toLowerCase().includes(searchTerm) &&
+      !data.entries[i].notes.toLowerCase().includes(searchTerm)
+    ) {
+      $currentLi.className = 'hidden';
+    } else {
+      $currentLi.className = '';
+    }
+  }
+});
+
+$clearSearchBtn.addEventListener('click', () => {
+  for (let i = 0; i < data.entries.length; i++) {
+    const $showLi = document.querySelector(
+      `li[data-entry-id="${data.entries[i].entryId}"]`,
+    );
+    if (!$showLi) throw new Error('$hideLi query failed');
+    $showLi.className = '';
+  }
```

Comment:
Same thing here, query for _all_ of the `li`s in the function and then loop through the _node list_.